backend/server/src/models/userModel.ts

```diff
@@ -1,9 +1,8 @@
-import mongoose, { Schema, type Types } from "mongoose";
-import { UserRole } from "./requestModel";
+import mongoose, { Schema } from "mongoose";
 import bcrypt from "bcryptjs";
 
 export interface User {
-  id: Types.ObjectId;
+  id: string;
```

Comment:
why are you storing id: string in stead of not using the default _id?

Reply:
Since _id is a virtual field. If you call `toObject()` on the user document, you wont have access to the _id field and it casts it to string anyways.

Reply:
And the `Types.ObjectId` is a string (a 24-character hexadecimal string).